client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -211,7 +212,7 @@ private boolean sendShuffleDataAsync(
                         .forEach(
                             blockId ->
                                 blockIdsSendFailTracker
-                                    .computeIfAbsent(blockId, id -> Lists.newArrayList())
+                                    .computeIfAbsent(blockId, id -> new CopyOnWriteArrayList())
```

Comment:
`CopyOnWriteArrayList` is suitable for the situation that we have more read operations than write operations. But we have more write operations than read operations. Maybe it will give us performance regression. If we want to use thread safe collection, maybe queue will be better.

Reply:
Ur right, replace it with `BlockingQueue`. Besides I revert the `blockIdsSendSuccessTracker` to the earlier code, since we don't track the success server.

Reply:
cc @yl09099